pkgs/by-name/ma/maltego/package.nix

```
@@ -0,0 +1,75 @@
+{ lib
+, stdenv
+, fetchzip
+, jre
+, giflib
+, gawk
+, makeBinaryWrapper
+, icoutils
+, copyDesktopItems
+, makeDesktopItem
+}:
+
+stdenv.mkDerivation rec {
+  pname = "maltego";
+  version = "4.6.0";
+  src = fetchzip {
+    url = "https://downloads.maltego.com/maltego-v4/linux/Maltego.v${version}.linux.zip";
+    hash = "sha256-q+1RYToZtBxAIDSiUWf3i/3GBBDwh6NWteHiK4VM1HY=";
+  };
+
+  postPatch = ''
+      substituteInPlace bin/maltego \
+            --replace /usr/bin/awk ${gawk}/bin/awk
+  '';
+
+    desktopItems = [
+    (makeDesktopItem {
+      name = pname;
+      desktopName = "Maltego";
+      exec = pname;
```

```suggestion
      exec = finalAttrs.meta.mainProgram;
```
pname is fine everywhere you are refering to a name but you should use mainProgram if you are referring to the binary, this also has nicer semantics if someone overrides the pname attribute for whatever reason.

---

Fixed